src/bin/wit-bindgen.rs

```diff
@@ -87,6 +92,16 @@ fn main() -> Result<()> {
             None => name.into(),
         };
         println!("Generating {:?}", dst);
+
+        if opt.check {
+            let prev = std::fs::read_to_string(&dst)
```

Comment:
I think this'll want to be `std::fs::read`, minus the `_to_string`, to handle some of the binary outputs of some generators,

Reply:
Fixed!